### DISCUSSION TITLE
Confused about accessing properties of a Release

### DISCUSSION THREAD

**Discussion by Simon-Dixon** [joalla/discogs_client#59] [Support]
I'm new to using the discogs_client (and relatively new to python), and I'm confused by an error I am getting.  I want to search for a Release and store its "id" so that I can look it up later via the id.  But when I look it up again, it is missing most of its properties.  What am I doing wrong?  Here's some code to reproduce the problem (d is my discogs_client.Client):

> results = d.search('Seven Days of Falling', label="215 Records", type='release')
> release1 = results[0]
> release2 = d.release(release1.id)
> print(release1)
> print(release2)
> print(type(release1))
> print(type(release2))
> print(x==release)
> print(x is release)
> print("Catno:", release1.data["catno"])
> print("Catno:", release2.data["catno"])

and the corresponding output:

<Release 8212964 'E.S.T. - Seven Days Of Falling'>
<Release 8212964 'Seven Days Of Falling'>
<class 'discogs_client.models.Release'>
<class 'discogs_client.models.Release'>
True
False
Catno: 215-2007

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-66-05fd4ff1a9d5> in <module>
     15 print(release1 is release2)
     16 print("Catno:", release1.data["catno"])
---> 17 print("Catno:", release2.data["catno"])
     18 print(release1.data.keys())
     19 print(release2.data.keys())

KeyError: 'catno'

Thanks!!

**Comment by JOJ0** [comment]
hi, on vaccation and mobile currently. maybe I'll try to reproduce in next days. Could you also wrap all your code and outputs into code block (triple backticks) It's hard to read at least on my mobile here.
thanks! but basically i dont see what could be the problem atm...

**Reply by Simon-Dixon** [reply]
Thanks Jojo,
No need to disturb your vacation for this!   I did a bit more reading through the source code and the thread about the project, and found that the catno is usually part of the label, so replacing
`release2.data["catno"]`
with
`release2.labels[0].fetch("catno"))`
has solved the immediate problem.  (I also discovered fetch() is superior to data["..."].)

But why the original release1 object contains a catno I don't understand.  My understanding now is that a single release can have multiple label/catno combinations, so release1.fetch("catno") is semantically wrong.  Presumably it worked because the search involved a label="...", which tied the result to a particular label.   

I'm not sure how to get triple backticks (manually?).   I was using the toolbar option "insert code" which I thought would do the right thing. 

Cheers
Simon
